generic/src/test/scala/neotypes/syntax/CypherQueryInterpolationSpec.scala

```diff
@@ -0,0 +1,114 @@
+      "p2" -> QueryParam(testClass.age)
+    )
+  }
+
+  it should "interpolation with a case class and extra args" in {
+    val testClass = TestClass("name", 33)
+    val bName = "b-name"
+
+    val (query, params, _) = c"CREATE (a: Test { $testClass }), (b: B { name: $bName })".build()
+
+    query shouldBe "CREATE (a: Test { name: $p1, age: $p2 }), (b: B { name: $p3 })"
+    params shouldBe Map(
+      "p1" -> QueryParam(testClass.name),
+      "p2" -> QueryParam(testClass.age),
+      "p3" -> QueryParam(bName)
+    )
+  }
+
+  it should "interpolation with a case class and extra args (concat queries)" in {
+    val testClass = TestClass("name", 33)
+    val bName = "b-name"
+
+    val (query, params, _) = (c"CREATE (a: Test { $testClass })," + c"(b: B { const: 'Const', name: $bName })").build()
+
+    query shouldBe "CREATE (a: Test { name: $p1, age: $p2 }), (b: B { const: 'Const', name: $p3 })"
+    params shouldBe Map(
+      "p1" -> QueryParam(testClass.name),
+      "p2" -> QueryParam(testClass.age),
+      "p3" -> QueryParam(bName)
+    )
+  }
+
+  it should "interpolation with case classes in a relationship" in {
+    val user = User("Joan", 20)
+    val cat = Cat("Waffles", 3)
+    val relationship = HasCatRelationship(2)
+
+    val (query, params, _) =
+      c"CREATE (u: User { $user })-[r: HAS_CAT { $relationship }]->(c: Cat { $cat }) RETURN r".build()
+
+    query shouldBe "CREATE (u: User { name: $p1, age: $p2 })-[r: HAS_CAT { friendsFor: $p3 }]->(c: Cat { tag: $p4, age: $p5 }) RETURN r"
+    params shouldBe Map(
+      "p1" -> QueryParam(user.name),
+      "p2" -> QueryParam(user.age),
+      "p3" -> QueryParam(relationship.friendsFor),
+      "p4" -> QueryParam(cat.tag),
+      "p5" -> QueryParam(cat.age)
+    )
+  }
+
+  it should "interpolation with a case classes and an extra property" in {
+    val user = User("Joan", 20)
+    val extraProp = 123
+
+    val (query, params, _) = c"CREATE (u: User { $user, extraProperty: $extraProp }) RETURN u".build()
+
+    query shouldBe "CREATE (u: User { name: $p1, age: $p2, extraProperty: $p3 }) RETURN u"
+    params shouldBe Map(
+      "p1" -> QueryParam(user.name),
+      "p2" -> QueryParam(user.age),
+      "p3" -> QueryParam(extraProp)
+    )
+  }
+
+  it should "interpolate null values just like optional values" in {
+    val optionalProperty: Option[String] = None
+    val nullProperty: String = null
+
+    val (query, params, _) = c"CREATE (u: User { op: ${optionalProperty}, np: ${nullProperty} }) RETURN u".build()
+
+    query shouldBe "CREATE (u: User { op: $p1, np: $p2 }) RETURN u"
+    params shouldBe Map(
+      "p1" -> QueryParam.NullValue,
+      "p2" -> QueryParam.NullValue
+    )
+  }
+
+  it should "interpolate literal nulls" in {
+    val (query, params, _) = c"CREATE (u: User { np: ${null} }) RETURN u".build()
+
+    query shouldBe "CREATE (u: User { np: $p1 }) RETURN u"
+    params shouldBe Map(
+      "p1" -> QueryParam.NullValue
+    )
+  }
+}
```

Comment:
These two are duplicated here and in the `core` spec, are they not?

Reply:
Yes, they're duplicated. I copied them to separately run tests for core module in Scala 3  like `++3.3.0 core/test`.

The change you pointed out in https://github.com/neotypes/neotypes/pull/666#discussion_r1306502360 is introduced for the same reason.

After all the modules are compiled in Scala 3, we don't need duplication.